be/tool/beck.c

```diff
@@ -898,6 +962,17 @@ static int scan(struct scanner *s)
 			lastoff  = s->s_off;
 			lastrecord = beck_builder.b_act;
 			lastdata = beck_builder.b_data;
+
+		}
+		/** save scanner offset if scanner and bnode queue's
+		 * are empty and scanner has progressed by delta bytes
+		 */
+		if( s->s_off - lastnvsaveoff >
+		    NV_OFFSET_SAVE_DELTA_IN_BYTES &&
+		    isqempty(&s->s_bnode_q)&&
+		    isqempty(s->s_q) ) {
```

Comment:
Is there any chance builder has not processed all the records and scanner queue is empty?

Reply:
when action added vs done counters are not matching we try to take lowest save bnode offset for that partitions and this offsetc as well as check will come into picture only when all partitions are idle.